src/lib/operators/join_hash.cpp

```diff
@@ -492,6 +493,7 @@ void probe_semi_anti(const RadixContainer<RightType>& radix_container,
         }
       } else if (mode == JoinMode::Anti) {
         // no hashtable on other side, but we are in Anti mode
+        pos_list_local.reserve(partition_end - partition_begin);
```

Comment:
In the `probe` method:
```
} else if (mode == JoinMode::Left || mode == JoinMode::Right) {
        /*
          We assume ...
          */

        for (size_t partition_offset = partition_begin; partition_offset < partition_end; ++partition_offset) {
          auto& row = partition[partition_offset];
          pos_list_left_local.emplace_back(NULL_ROW_ID);
          pos_list_right_local.emplace_back(row.row_id);
        }
      }
```
, the `pos_list_left_local` and `pos_list_left_local` could reserve `partition_end - partition_begin` as well. 

I also thought about how to reserve for `pos_list_left_local` and `pos_list_left_local` if a hashtable for the current partition was found. But I didn't find a good way yet. This would be good for TPC-H 9.

Reply:
I added two reserve() calls. It didn't change the iterations/s though.

Reply:
Do you have any idea how large these position lists get for left and right joins in TPC-H?

Reply:
It looks like TPC-H 13 is the only query with a left join. The join output has ~500k rows with scale factor 1. So the position list for one partition has ~1k RowIDs (when using 9 radix bits --> 512 partitions)

Reply:
The pre-allocation of one thousand rows is probably not enough to show any significant performance improvement. However, it should not hurt either.

Reply:
Yes, and this only reserves if there are no values in the partition of the smaller input. Maybe that happens not very often.